services/src/functional-tests/README.md

```diff
@@ -57,75 +71,52 @@ For example:
-For virtual (mininet) topology this definition will serve as a guide for creating a virtual topology.
-
-The topology definition format:
-```
-switches:
-    - name: ofsw1
-      dp_id: 00:00:00:00:00:00:00:01
-      of_version: OF_13
-      status: active
-      out_ports:
-        - port: 10
-          vlan_range: 101..150
-
-    - name: ofsw2
-      dp_id: 00:00:00:00:00:00:00:02
-      of_version: OF_13
-      status: active
-      out_ports:
-        - port: 10
-          vlan_range: 101..150
-isls:
-    - src_switch: ofsw1
-      src_port: 2
-      dst_switch: ofsw2
-      dst_port: 4
-      max_bandwidth: 36000000
-    - src_switch: ofsw1
-      src_port: 1
-      dst_switch: ofsw2
-      dst_port: 1
-      a_switch:
-        in_port: 49
-        out_port: 50
-      max_bandwidth: 36000000
-    - src_switch: ofsw1
-      src_port: 5
-      max_bandwidth: 36000000
-      a_switch:
-        in_port: 20
-traffgens:
-    - name: tg1
-      iface: eth0
-      control_endpoint: http://localhost:4041
-      switch: ofsw1
-      switch_port: 10
-      status: active
-    - name: tg2
-      iface: eth0
-      control_endpoint: http://localhost:4042
-      switch: ofsw2
-      switch_port: 10
-      status: active
-
-traffgen_config:
-    address_pool_base: 0.0.0.0
-    address_pool_prefix_len: 20
-```
-
-### Kilda configuration
-The tests require Kilda configuration provided. See `kilda.properties.example`.
+## Best Practices
+- Don't be too laconic when naming a test. Specify what behavior is being tested instead
+of what actions are being took.  
+  - Good:
+    - "Unable to delete meter with invalid id";
+    - "Flow in 'Down' status is rerouted when discovering a new ISL". 
+  - Bad:
+    - "Delete meter with invalid id";
+    - "Discover new ISL while flow is down".
+- Please provide a comprehensive comment for *every* given-when-then block, so that it forms a valid readable
+test case at the end.
+- Add a blank line between given-when-then blocks.
+- Make your best on cleaning up after your test. Make sure your test does not influence subsequent tests:
+  - revert any changes made to the system: topology, database, switches etc.;
+  - take care of doing all the proper waits after your test has finished to be sure that all your reverts actually took
+  place.
+- Make sure your test is stable. Run it multiple times before committing. Cases that can often be unstable
+(list is not exhaustive, just for an example purpose):
+  - flow creation, deletion and reroute may require additional waits for flow rules to be actually installed
+  on switches, since floodlight needs some time to install them, while Northbound responds that everything's done.
+  Thus knocking out switch right after flow create/delete command is a dangerous operation;
+  - switches knockout/revive operations should have proper waits to ensure that their actual status has changed.
+  Same for ISLs;
```

Comment:
According to your style (which I like), there should be a dot here instead of ";".

Reply:
fxd